conda_smithy/lint_recipe.py

```diff
@@ -475,6 +475,8 @@ def lintify(meta, recipe_dir=None, conda_forge=False):
     # 22: Single space in pinned requirements
     for section, requirements in requirements_section.items():
         for requirement in requirements or []:
+            # rather than adding to utils.render_meta_yaml() we could
+            # handle requirement is None by ignoring it
```

Comment:
I don't completely understand how `{{ blas_impl }}` gets rendered as `None` given the implementation of `conda_smithy.utils.NullUndefined`.  However, thinking about this more, it probably makes sense to continue whitelisting variables in `utils.render_meta_yaml` rather than ignoring `requirement is None` here.